### PR TITLE
[18EUS] Issue/redeem, EMR train buying, bankruptcy, and end game

### PR DESCRIPTION
**Pull request by crericha**
Some other misc too.

### CODE REVIEW

File: lib/engine/game/base.rb [hunk 1]
```diff
@@ -248,6 +248,7 @@ class Base
       DISCARDED_TRAINS = :discard # discard or remove
       DISCARDED_TRAIN_DISCOUNT = 0 # percent
       CLOSED_CORP_TRAINS_REMOVED = true
+      CLOSED_CORP_TOKENS_REMOVED = true
```
Comment (outsidePasser) [inline lib/engine/game/base.rb:252]: Isn't this just a dup of the line above?

File: lib/engine/game/g_18_eus/game.rb [hunk 2]
```diff
@@ -99,8 +110,8 @@ class Game < Game::Base
         ].freeze
 
         TRAINS = [
-          { name: '2', distance: 2, price: 100, rusts_on: '4', num: 20 },
-          { name: '2+', distance: 2, price: 100, obsolete_on: '4', num: 10 },
+          { name: '2', distance: 2, price: 100, rusts_on: '4', num: 2 },
```
Comment (outsidePasser) [inline lib/engine/game/g_18_eus/game.rb]: Typo?  We need more then 2 2-trains...

File: lib/engine/game/g_18_eus/game.rb [hunk 3]
```diff
@@ -290,20 +318,29 @@ def stock_round
 
         def operating_round(round_num)
           Engine::Round::Operating.new(self, [
-            Engine::Step::Bankrupt,
+            G18EUS::Step::Bankrupt,
             Engine::Step::Exchange,
+            Engine::Step::DiscardTrain,
             G18EUS::Step::SpecialTrack,
-            Engine::Step::BuyCompany,
```
Comment (outsidePasser) [inline lib/engine/game/g_18_eus/game.rb]: I put a "skip" in for this step - will need to change that to acquire.  I can do that after your gets merged in.

File: lib/engine/game/g_18_eus/game.rb [hunk 4]
```diff
@@ -469,8 +520,42 @@ def company_status_str(company)
           return "Bid box #{index + 1}" if index && index < self.class::BIDDING_BOX_PRIVATE_COUNT
         end
 
-        def bidding_token_per_player
-          self.class::BIDDING_BOX_PRIVATE_COUNT
+        def revenue_for(route, stops)
```
Comment (outsidePasser) [inline lib/engine/game/g_18_eus/game.rb:525]: Will need to merge my version of this in

File: lib/engine/game/g_18_eus/step/buy_train.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/buy_train'
+
+module Engine
+  module Game
+    module G18EUS
+      module Step
+        class BuyTrain < Engine::Step::BuyTrain
+          def actions(entity)
```
Comment (outsidePasser) [inline lib/engine/game/g_18_eus/step/buy_train.rb:13]: Will need to merge in my "skip_bny" include

**Review by outsidePasser — APPROVED** [review]
Looks good, just a couple comments

File: lib/engine/game/base.rb (hunk 1 above)
Comment (crericha) [reply lib/engine/game/base.rb:252]: Yeah, bad merge.

File: lib/engine/game/g_18_eus/game.rb (hunk 2 above)
Comment (crericha) [reply lib/engine/game/g_18_eus/game.rb]: Testing code that I forgot to revert. Good catch.

File: lib/engine/game/g_18_eus/game.rb (hunk 4 above)
Comment (crericha) [reply lib/engine/game/g_18_eus/game.rb:525]: I'll handle merging the two.

Should bidding_token_per_player still be there? Or did I move it elsewhere?

File: lib/engine/game/g_18_eus/step/buy_train.rb (hunk 5 above)
Comment (crericha) [reply lib/engine/game/g_18_eus/step/buy_train.rb:13]: Yes. I'll handle after your PR comes in.

**Pull request merged**